src/libs/ajax/DatasetBuilder.ts

```diff
@@ -148,13 +174,51 @@ const getDummyConcepts = async (parent: Concept): Promise<GetConceptsResponse> =
   };
 };
 
-export const DatasetBuilder = (): DatasetBuilderContract => ({
-  retrieveDataset: async (datasetId) => {
-    return await Ajax()
-      .DataRepo.dataset(datasetId)
-      .details([datasetIncludeTypes.SNAPSHOT_BUILDER_SETTINGS, datasetIncludeTypes.PROPERTIES]);
-  },
-  getConcepts: (parent: Concept) => Promise.resolve(getDummyConcepts(parent)),
-  requestAccess: (_request) => Promise.resolve(),
-  getParticipantCount: (_request) => Promise.resolve(100),
-});
+export const DatasetBuilder = (): DatasetBuilderContract => {
+  return {
+    retrieveDataset: async (datasetId) => {
+      return await Ajax()
+        .DataRepo.dataset(datasetId)
+        .details([datasetIncludeTypes.SNAPSHOT_BUILDER_SETTINGS, datasetIncludeTypes.PROPERTIES]);
+    },
+    getProgramDataStatistics: async (datasetId, programDataOption) => {
+      switch (programDataOption.kind) {
+        case 'list':
+          return {
+            name: programDataOption.name,
+            kind: programDataOption.kind,
+            values: _.map(
+              (num) => ({
+                id: num,
+                name: `${programDataOption.name} ${num}`,
+              }),
+              [0, 1, 2, 3, 4]
+            ),
+          };
+        case 'range':
+          const statistics = await Ajax()
+            .DataRepo.dataset(datasetId)
+            .lookupDatasetColumnStatisticsById(programDataOption.tableName, programDataOption.columnName);
+
+          switch (statistics.dataType) {
+            case 'integer' || 'int64' || 'numeric':
+              return {
+                name: programDataOption.name,
+                kind: programDataOption.kind,
+                min: (statistics as ColumnStatisticsIntOrDoubleModel).minValue,
```

Comment:
`as ColumnStatisticsIntOrDoubleModel` wouldn't be needed here if `dataType` is added as a discriminator field to `ColumnStatisticsIntOrDoubleModel`, something like:
```
export interface ColumnStatisticsModel {
  dataType: 'integer' | 'int64' | 'numeric' | 'text' | 'string';
}

export interface ColumnStatisticsIntOrDoubleModel extends ColumnStatisticsModel {
  dataType: 'integer' | 'int64' | 'numeric';
  minValue: number;
  maxValue: number;
}

export interface ColumnStatisticsTextModel extends ColumnStatisticsModel {
  dataType: 'text' | 'string';
  values: ColumnStatisticsTextValue[];
}
```